### ISSUE TITLE
Error when saving email settings

### ISSUE THREAD

**Issue by mpaluchowski** [fisharebest/webtrees#3547]
I'm running version 2.0.9. When I go to the "Sending email" screen, change the "Sender name" entry and click "Save", the page load takes a long, long time and then throws an error below. The changed name doesn't get saved in the database (`*site_setting` table).

```
Fatal error: Uncaught ErrorException: Trying to get property 'name' of non-object in /home/nethut/public_html/genealogy.nethut.pl/resources/views/layouts/default.phtml:71 Stack trace:
#0 /home/nethut/public_html/genealogy.nethut.pl/resources/views/layouts/default.phtml(71): Fisharebest\Webtrees\Webtrees::Fisharebest\Webtrees\{closure}(8, 'Trying to get p...', '/home/nethut/pu...', 71, Array)
#1 /home/nethut/public_html/genealogy.nethut.pl/app/View.php(189): include('/home/nethut/pu...')
#2 /home/nethut/public_html/genealogy.nethut.pl/app/View.php(286): Fisharebest\Webtrees\View->render()
#3 /home/nethut/public_html/genealogy.nethut.pl/app/Helpers/functions.php(203): Fisharebest\Webtrees\View::make('layouts/default', Array)
#4 /home/nethut/public_html/genealogy.nethut.pl/app/Http/ViewResponseTrait.php(58): view('layouts/default', Array)
#5 /home/nethut/public_html/genealogy.nethut.pl/app/Http/Middleware/HandleExceptions.php(172): Fisharebest\Webtrees\Http\Middleware\HandleExceptions->viewResponse('components/aler...', Arra in /home/nethut/public_html/genealogy.nethut.pl/resources/views/layouts/default.phtml on line 71
```

**Comment by mpaluchowski** [comment]
I also noticed, that the initial value for `SMTP_ACTIVE` set by `SeedSiteSettingTable.php` is `internal`, but the only available options now are `sendmail` and `external`.

**Comment by fisharebest** [comment]
This is an error in the error-handler.

So, the original error message is lost.

If it happens after a long time, then I guess that (a) you also selected the "send a test email" option, and (b) there was an error sending the mail.

But this case should be handled OK.  Without the original error message, it is difficult to investigate.

Do you still get the error with the latest version of webtrees?

**Comment by mpaluchowski** [comment]
Seems to be working fine now. Yes, I did check the "send a test email" option to trigger that.

When my sendmail settings are correct (as per #3548) then it updates the data and sends the test message.
If I set an invalid path to `sendmail`, then it reports a simple error of the message not having been sent.
Only when I set `sendmail_command="/usr/sbin/sendmail -bs"` (where my correct setting is `-t`), then the submitting the form keeps loading indefinitely.
